docs/how-to-guides/config-settings.rst

```diff
@@ -9,41 +9,53 @@ Use build config settings
 *************************
 
 Build config settings are settings you can pass to ``meson-python`` when
-building your project. They can be use to customize the build in some aspect.
+building your project. They can be use to customize the build in some
+aspect.
 
-The way you specify them depends on which tool you are using, here are the most
-popular ones.
+Several Python build front-ends exist, with different ways to pass
+configuration settings to the build back-end. The most popular are
+``pypa/build``, which uses the ``-C`` command line option, and ``pip``,
+which uses the ``--config-settings`` option. For example:
 
+.. tab-set::
 
-`pypa/build`_
-=============
+    .. tab-item:: pypa/buid
+        :sync: key_pypa_build
 
-On `pypa/build`_, they can be used via the ``-C``/``--config-setting`` option.
+	.. code-block:: console
 
+	   $ python -m build \
+               -Csetup-args="-Doption=true" \
+               -Csetup-args="-Dvalue=1" \
+               -Ccompile-args="-j6"
 
-.. code-block:: console
+    .. tab-item:: pip
+        :sync: key_pip
 
-   $ python -m build -Ckey=value
+	.. code-block:: console
 
+	   $ python -m pip wheel . \
+               --config-settings=setup-args="-Doption=disable" \
+               --config-settings=compile-args="-j6"
 
-Please check the `pypa/build documentation`_ for more information.
+This examples use the ``python -m pip wheel`` command to build a Python
+wheel that can be later installed or distributed. To build a package and
+immediately install it, just replace ``wheel`` with ``install``.
 
+See :ref:`how-to-guides-meson-args` for more examples. Refer to the
+`pypa/build documentation`_ or to the `pip documentation`_ for more
+information.
 
-pip_
-====
-
-On `pip`_, they can be used via the ``--config-settings`` option.
-
-
-.. code-block:: console
-
-   $ pip install . --config-settings key=value
-
-
-Please check the `pip documentation`_ for more information.
-
-
-.. _pypa/build: https://github.com/pypa/build
 .. _pypa/build documentation: https://pypa-build.readthedocs.io/en/stable/
-.. _pip: https://github.com/pypa/pip
 .. _pip documentation: https://pip.pypa.io/
+
+Please note that, while ``pypa/build`` concatenates arguments for the same
+key passed to the ``-C`` option, ``pip`` up to version 23.0.1 does not offer
+any way to set a build config setting to a list of strings: later values for
+the same key passed to ``--config-settings`` override earlier ones. This
+effectively limits the number of options that can be passed to each command
+invoked in the build process to one. This limitation is tracked in `pip
+issue #11681`_. This limitation should be removed in the next version of
+pip.
```

Comment:
I think this should be an admonition, also, the last two sentences can be combined. It might also make sense to split into two paragraphs if we're putting it in an admonition.

```suggestion
.. admonition:: pip <= 23.0.1 limitation
   :class: caution
   
   Please note that, while `pypa/build`_ concatenates arguments for the same
   key passed to the ``-C`` option, `pip`_ up to version 23.0.1 does not offer
   any way to set a build config setting to a list of strings: later values for
   the same key passed to ``--config-settings`` override earlier ones.
   
   This effectively limits the number of options that can be passed to each
   command invoked in the build process to one. This limitation is tracked
   in `pip issue #11681`_ and should be removed in the next version of pip.
```